zirgen/Conversions/Typing/ZhlComponent.cpp

```diff
@@ -1540,6 +1576,7 @@ void LoweringImpl::buildZeroInitialize(Value toInit) {
     Value zeroDistance = builder.create<arith::ConstantOp>(loc, builder.getIndexAttr(0));
     auto reload = builder.create<ZStruct::LoadOp>(loc, valType, unwrap, zeroDistance);
     builder.create<Zll::EqualZeroOp>(loc, reload);
+    break;
```

Comment:
Is this intentional? With the break here, I cannot see how this loop will ever execute more than one iteration, which makes me wonder why it is a loop at all.

Reply:
Yes, this is intentional, although I agree perhaps it's better to remove the loop in it's entirety as you suggest, it's a bit unclear as it is.  Basically, when we zero initialize the arguments, as long as the 'count' (which is always the first field) is zero, the other elements are irrelevant.  This is useful to allow, so that we can 'alias' other elements to non-zero values.